swap/src/protocol/bob/swap.rs

```diff
@@ -221,31 +225,59 @@ async fn next_state(
 
             let tx_lock_status = bitcoin_wallet.subscribe_to(state.tx_lock.clone()).await;
 
-            if let ExpiredTimelocks::None { .. } = state.expired_timelock(bitcoin_wallet).await? {
-                let watch_request = state.lock_xmr_watch_request(lock_transfer_proof);
+            // Check if the cancel timelock has expired => we cancel the swap
+            let ExpiredTimelocks::None { .. } = state.expired_timelock(bitcoin_wallet).await?
```

Comment:
Imo this syntax is worse than what was used before. It's not entirely clear when this codeblock will be executed at first sight. You need to know which other enum variants exist